{"version":3,"sources":["authError.js"],"names":["AuthError","response","status","json","text","error","description","error_description","BaseError"],"mappings":";;;;;;;;;;;;;;;AAAA;;IAEqBA,S;;;AACnB,qBAAYC,QAAZ,EAAsB;AAAA;;AAAA;AAAA,QACZC,MADY,GACgBD,QADhB,CACZC,MADY;AAAA,yBACgBD,QADhB,CACJE,IADI;AAAA,QACJA,IADI,+BACG,EADH;AAAA,QACOC,IADP,GACgBH,QADhB,CACOG,IADP;AAAA,QAEZC,KAFY,GAE8BF,IAF9B,CAEZE,KAFY;AAAA,QAEcC,WAFd,GAE8BH,IAF9B,CAELI,iBAFK;AAGpB,+GACEF,KAAK,IAAI,qBADX,EAEEC,WAAW,IAAIF,IAAf,IAAuB,eAFzB;AAIA,UAAKD,IAAL,GAAYA,IAAZ;AACA,UAAKD,MAAL,GAAcA,MAAd;AARoB;AASrB;;;EAVoCM,kB","sourcesContent":["import BaseError from '../utils/baseError';\n\nexport default class AuthError extends BaseError {\n  constructor(response) {\n    const { status, json = {}, text } = response;\n    const { error, error_description: description } = json;\n    super(\n      error || 'a0.response.invalid',\n      description || text || 'unknown error'\n    );\n    this.json = json;\n    this.status = status;\n  }\n}\n"]}